ta/remoteproc/remoteproc_core.c

```diff
@@ -0,0 +1,1212 @@
+ *                /       ||   TLV     ||   Information used to authenticate the
+ *  +-----------+/        ||           ||   firmware, stored in
+ *  |  Trailer  |         |+-----------+|   Type-Length-Value format.
+ *  +-----------+\        |+-----------+|
+ *                \       || Platform  ||   Specific platform information,
+ *                 \      ||   TLV     ||   stored in Type-Length-Value format.
+ *                  \     |+-----------+|
+ *                   -----+-------------+
+ *
+ *  TLV and platform TLV chunks:
+ *
+ *                        +-------------+
+ *                        |    Magic    |
+ *                        +-------------+
+ *                        +-------------+
+ *                        |    size     |  size of the TLV payload
+ *                        +-------------+
+ *                        +-------------+
+ *                        |     TLV     |  TLV structures
+ *                        |   payload   |
+ *                        +-------------+
+ */
+
+/* Firmware state */
+enum remoteproc_state {
+	REMOTEPROC_OFF,
+	REMOTEPROC_LOADED,
+	REMOTEPROC_STARTED,
+};
+
+#define RPROC_HDR_MAGIC		0x3543A468 /* random value */
+#define HEADER_VERSION		1
+
+/* Supported signature algorithm */
+enum remoteproc_sign_type {
+	RPROC_RSASSA_PKCS1_v1_5_SHA256 = 1,
+	RPROC_ECDSA_SHA256 = 2,
+};
+
+enum remoteproc_img_type {
+	REMOTEPROC_ELF_TYPE = 1,
+	REMOTEPROC_INVALID_TYPE = 0xFF
+};
+
+/* remoteproc_tlv structure offsets */
+#define RPROC_TLV_LENGTH_OF	U(0x02)
+#define RPROC_TLV_VALUE_OF	U(0x04)
+
+#define TLV_INFO_MAGIC		U(0x6907)
+#define PLAT_TLV_INFO_MAGIC	U(0x6908)
+
+/* TLV types */
+#define RPROC_TLV_SIGNTYPE	U(0x01)
+#define RPROC_TLV_HASHTYPE	U(0x02)
+#define RPROC_TLV_NUM_IMG	U(0x03)
+#define RPROC_TLV_IMGTYPE	U(0x04)
+#define RPROC_TLV_IMGSIZE	U(0x05)
+#define RPROC_TLV_HASHTABLE	U(0x10)
+#define RPROC_TLV_PKEYINFO	U(0x11)
+
+#define RPROC_TLV_SIGNTYPE_LGTH U(1)
+
+#define LE16_TO_CPU(x) (((x)[1] << 8) | (x)[0])
+#define LE32_TO_CPU(x) (((x)[3] << 24) | ((x)[2] << 16) | \
+			((x)[1] << 8)  | (x)[0])
+
+#define U64_ADD_TO_ALIGN(x) ((x) + sizeof(uint64_t) - ((x) % sizeof(uint64_t)))
+#define U64_ALIGN_SZ(x) ((x) % (sizeof(uint64_t)) ? U64_ADD_TO_ALIGN(x) : (x))
+
+/*
+ * struct remoteproc_tlv - Type-Length-Value structure
+ * @type: type of data
+ * @length: size of the data.
+ * @value: pointer to the data.
+ */
+struct remoteproc_tlv {
+	uint16_t type;
+	uint16_t length;
+	uint8_t value[0];
+};
+
+/*
+ * struct remoteproc_tlv_hr - type length data sutructure
+ * @magic: magic value of the TLV chunk
+ * @size: total size of the TLV values.
+ */
+struct remoteproc_tlv_hr {
+	int32_t magic;
+	int32_t size;
+};
+
+/*
+ * struct remoteproc_segment - program header with hash structure
+ * @phdr: program header
+ * @hash: hash associated to the program segment.
+ */
+struct remoteproc_segment {
+	Elf32_Phdr phdr;
+	unsigned char hash[TEE_SHA256_HASH_SIZE];
```

Comment:
s/char/uint8_t/
ditto at all other place where hash is declare as a char array.